src/plugins/global-settings/color-schemes/index.php

```diff
@@ -0,0 +1,505 @@
+					'selectors' => array(
+						'desktop' => ':root'
+					)
+				),
+				array(
+					'key' => $background_default,
+					'mode' => 'background',
+					'selectors' => array(
+						'desktop' => '.stk-block-background',
+						'desktopParentHover' => ':where(.stk-hover-parent:hover) .stk-block-background'
+					)
+				),
+				array(
+					'key' => $container_default,
+					'mode' => 'container',
+					'selectors' => array(
+						'desktop' => '.stk-container:where(:not(.stk--no-background))',
+						'desktopParentHover' => array( '.stk-container:where(:not(.stk--no-background):hover)', ':where(.stk-hover-parent:hover) .stk-container:where(:not(.stk--no-background))')
+					)
+				)
+			);
+
+			$block_color_schemes = array(
+				'background' => array(),
+				'container' => array(),
+			);
+
+			// generate selectors for all schemes in background and container mode
+			foreach ($this->color_schemes as $key => $_ ) {
+				$block_color_schemes['background'][] = array(
+					'key' => $key,
+					'mode' => 'background',
+					'selectors' => array(
+						'desktop' => ".background-$key",
+						'desktopParentHover' => ":where(.stk-hover-parent:hover) .background-$key"
+					)
+				);
+				$block_color_schemes['container'][] = array(
+					'key' => $key,
+					'mode' => 'container',
+					'selectors' => array(
+						'desktop' => ".container-$key",
+						'desktopParentHover' => array(".container-$key:where(:hover)",":where(.stk-hover-parent:hover) .container-$key")
+					)
+				);
+			}
+
+			foreach( $default_color_schemes as $scheme ) {
+				$styles = $this->generate_color_scheme_styles( $styles, $scheme );
+			}
+
+			$generated_css = wp_style_engine_get_stylesheet_from_css_rules( $styles );
+			if ( $generated_css != '' ) {
+				$current_css .= "\n/* Global Color Schemes */\n";
+				$current_css .= $generated_css;
+			}
+
+			foreach( $block_color_schemes as $mode => $block_schemes ) {
+				foreach( $block_schemes as $scheme ) {
+					$styles = $this->generate_color_scheme_styles( array(), $scheme );
+					$generated_css = wp_style_engine_get_stylesheet_from_css_rules( $styles );
+					if ( $generated_css != '' ) {
+						$scheme_key = $scheme[ 'key' ];
+						$current_css .= "\n/* Global Color Schemes ($mode-$scheme_key) */\n";
+						$current_css .= $generated_css;
+					}
+				}
+			}
+
+			return apply_filters( 'stackable_frontend_css' , $current_css );
+		}
+
+		/**
+		 * This converts the Color Schemes from the database to an associative array where
+		 * the key is the color scheme slug and the value is the color scheme array itself.
+		 *
+		 * This allows us to easily check if the color scheme exists
+		 * and retrieve the color scheme by slug
+		 *
+		 * @param Array 	$schemes_array
+		 * @return Array
+		 */
+		public function convert_to_assoc_array( $schemes_array ) {
+			$schemes = array();
+
+			foreach( $schemes_array as $scheme ) {
+				$schemes[ $scheme['key'] ] = $scheme[ 'colorScheme' ];
+			}
+
+			return $schemes;
```

Comment:
We can get rid of the foreach here:
```suggestion
			return array_column( $schemes_array, 'colorScheme', 'key' );
```